### PR TITLE
More checks in poll

### PR DESCRIPTION
**Pull request by guidow**

### CODE REVIEW

File: pyfarm/scheduler/tasks.py [hunk 1]
```diff
@@ -504,6 +505,12 @@ def poll_agent(self, agent_id):
                     agent.hostname, agent.id, status_response.status_code))
         status_json = status_response.json()
 
+        if UUID(status_json["agent_id"]) != agent_id:
+            logger.error("Wrong agent reached under %s. Expected id %s, got %s",
+                         agent.api_url(), agent_id, status_json["agent_id"])
+            raise ValueError("Wrong agent_id on polling. Excepted: %s. Got %s" %
```
Comment (opalmer) [inline pyfarm/scheduler/tasks.py]: typo: Excepted -> Expected

File: pyfarm/scheduler/tasks.py [hunk 2]
```diff
@@ -504,6 +505,12 @@ def poll_agent(self, agent_id):
                     agent.hostname, agent.id, status_response.status_code))
         status_json = status_response.json()
 
+        if UUID(status_json["agent_id"]) != agent_id:
+            logger.error("Wrong agent reached under %s. Expected id %s, got %s",
```
Comment (opalmer) [inline pyfarm/scheduler/tasks.py]: typo: Excepted -> Expected

File: pyfarm/scheduler/tasks.py [hunk 3]
```diff
@@ -565,6 +572,22 @@ def poll_agent(self, agent_id):
                          "to have. Registering task pusher", agent.hostname)
             send_tasks_to_agent.delay(agent_id)
 
+        if set(present_task_ids) - set(assigned_task_ids):
```
Comment (opalmer) [inline pyfarm/scheduler/tasks.py]: Could we do the set subtraction once?

File: pyfarm/scheduler/tasks.py (hunk 2 above)
Comment (guidow) [inline pyfarm/scheduler/tasks.py]: Not on this line.

Comment (opalmer) [inline pyfarm/scheduler/tasks.py]: Yeah sorry my bad....

**Pull request merged**